lib/modules/dosomething/dosomething_signup/dosomething_signup.module

```diff
@@ -584,8 +602,13 @@ function dosomething_signup_user_signup($nid, $account = NULL, $source = NULL) {
  *   Details about the node that the signup was made on.
  * @param int $opt_in
  *   The opt-in path to use for opt-in.
+ * @param array $options
+ *   - transactionals
+ *   (optional) A flag to disable sending transactional messages.
+ *   - source
+ *   (optional) A idendify name of the original source that requested the campaign signup.
  */
-function dosomething_signup_mbp_request($account, $node, $opt_in) {
+function dosomething_signup_mbp_request($account, $node, $opt_in, $options) {
```

Comment:
Let's declare $options as an empty array everywhere, like you did in `dosomething_signup_third_party_subscribe()`.